pandas/tests/base/test_unique.py

```diff
@@ -105,3 +105,21 @@ def test_nunique_null(null_obj, index_or_series_obj):
         num_unique_values = len(obj.unique())
         assert obj.nunique() == max(0, num_unique_values - 1)
         assert obj.nunique(dropna=False) == max(0, num_unique_values)
+
+
+@pytest.mark.parametrize(
+    "idx_or_series_w_bad_unicode", [pd.Index(["\ud83d"] * 2), pd.Series(["\ud83d"] * 2)]
+)
+def test_unique_bad_unicode(idx_or_series_w_bad_unicode):
+    # regression test for #34550
+    obj = idx_or_series_w_bad_unicode
+    result = obj.unique()
+
+    # dict.fromkeys preserves the order
+    unique_values = list(dict.fromkeys(obj.values))
```

Comment:
You can remove this and then simply pd.Index(["\ud38d"], dtype=object) below

Reply:
Sorry I missed that part of your comment, fixed now :)